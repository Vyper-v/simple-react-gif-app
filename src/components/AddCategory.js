import React,{useState} from 'react';
import PropTypes from 'prop-types';


function AddCategory({ setCategories }) {
  const [value, setValue] = useState('');
  function handleInputChange(e) {
    setValue(e.target.value);
  }

  function handleSubmit(e) {
    e.preventDefault();
    if (value !== ''){
      setCategories(categs => [value,...categs]);
      setValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit} className='flex flex-col'>
      <input className="p-1 outline-1" type="text" placeholder='Type something...' value={value} onChange={handleInputChange} />
    </form>
  );
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};


export default AddCategory;
