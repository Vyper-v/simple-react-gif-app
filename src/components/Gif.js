import React from 'react';
import PropTypes from 'prop-types';

const Gif = ({title,url}) => {
  return <div className='flex flex-col justify-center items-baseline rounded overflow-hidden shadow shadow-lime-500 bg-lime-200 text-lime-700 transition-all'>
    <img className='object-cover w-full h-[20rem]' src={url} alt={title} />
    <span className='text-lg p-4 flex-auto'>{title}</span>
  </div>;
};

Gif.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Gif;
