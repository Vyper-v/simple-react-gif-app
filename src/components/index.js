export {default as Gif} from './Gif'
export {default as GifLayout} from './GifLayout'
export {default as AddCategory} from './AddCategory'