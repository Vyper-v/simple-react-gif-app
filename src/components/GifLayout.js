import PropTypes from 'prop-types';

import Gif from './Gif';
import { useFetchGifs } from '../hooks/useFechGifs';
import './GifLayout.css'

const GifLayout = ({category}) => {
  const {data:images,loading} = useFetchGifs(category);

  return (
  <div className='grid autoLayout gap-2'>
    { loading && <p className='text-center'>Loading...</p> }
    { images.map( (img) =>  <Gif key={img.id} {...img}/> ) }
  </div>
  );
};

GifLayout.propTypes = {
  category: PropTypes.string.isRequired
};

export default GifLayout;
