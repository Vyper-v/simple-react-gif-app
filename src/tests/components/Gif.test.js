import {Gif} from '../../components/index';
// import getGifs from '../../helpers/getGifs';
import {shallow} from 'enzyme';

describe('Gif Component tests', () => {
  const title = 'Test';
  const url = 'https://picsum.photos/200/300';
  const wrapper = shallow(<Gif title={title} url={url} />);
  
  

  test('should be rendered correctly', async () => {
    // const category = 'Overwatch';
    // const gifData = await getGifs(category)
    expect(wrapper).toMatchSnapshot();
  });

  test('should have span with text', () => {
    const span = wrapper.find('span').text().trim();
    expect(span).toBe(title);
  });

  test('should have image', () => {
    const img = wrapper.find('img');
    const {src,alt} = img.props();
  
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should have a class "flex"', () => {
    const {className} = wrapper.props();
    expect(className.includes("flex")).toBe(true);
  });
  
  
  
  
});
