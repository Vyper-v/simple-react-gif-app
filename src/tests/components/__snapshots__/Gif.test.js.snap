// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Component tests should be rendered correctly 1`] = `
<div
  className="flex flex-col justify-center items-baseline rounded overflow-hidden shadow shadow-lime-500 bg-lime-200 text-lime-700 transition-all"
>
  <img
    alt="Test"
    className="object-cover w-full h-[20rem]"
    src="https://picsum.photos/200/300"
  />
  <span
    className="text-lg p-4 flex-auto"
  >
    Test
  </span>
</div>
`;
